From 8a51e9551f4dcb89ec57a713f9ee153fc961a64d Mon Sep 17 00:00:00 2001
From: Bruno Jesus <00cpxxx@gmail.com>
Date: Tue, 1 Dec 2015 19:51:20 +0800
Subject: shlwapi/tests: Test NULL handle duplication in SHMapHandle()

Related to https://bugs.winehq.org/show_bug.cgi?id=36838

Not sure when Olivier Dierick will be back to this so I'm sending the
test I made.

Signed-off-by: Bruno Jesus <00cpxxx@gmail.com>
---
 dlls/shlwapi/tests/ordinal.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/shlwapi/tests/ordinal.c b/dlls/shlwapi/tests/ordinal.c
index bd3d46a..df07df4 100644
--- a/dlls/shlwapi/tests/ordinal.c
+++ b/dlls/shlwapi/tests/ordinal.c
@@ -550,6 +550,12 @@ static void test_alloc_shared_remote(DWORD procid, HANDLE hmem)
     ok(ret, "SHUnlockShared failed: %u\n", GetLastError());
 
     /* test SHMapHandle */
+    SetLastError(0xdeadbeef);
+    hmem2 = pSHMapHandle(NULL, procid, GetCurrentProcessId(), 0, 0);
+    ok(hmem2 == NULL, "expected NULL, got new handle\n");
+todo_wine
+    ok(GetLastError() == 0xdeadbeef, "last error should not have changed, got %u\n", GetLastError());
+
     hmem2 = pSHMapHandle(hmem, procid, GetCurrentProcessId(), 0, 0);
 
     /* It seems like Windows Vista/2008 uses a different internal implementation
-- 
2.6.4

